src/bindings/python/PyGpuShaderDesc.cpp

```diff
@@ -285,11 +327,12 @@ void bindPyGpuShaderDesc(py::module & m)
                 const char * samplerName = nullptr;
                 unsigned width, height;
                 GpuShaderDesc::TextureType channel;
+                GpuShaderDesc::TextureDimensions dimensions;
                 Interpolation interpolation;
-                it.m_obj->getTexture(i, textureName, samplerName, width, height, channel, 
+                it.m_obj->getTexture(i, textureName, samplerName, width, height, channel, dimensions,
                                      interpolation);
 
-                return { textureName, samplerName, width, height, channel, interpolation,
+                return { textureName, samplerName, width, height, channel, dimensions, interpolation,
```

Comment:
Assuming it's going to break existing Python code but not sure how to do it differently, same for `__next__`.

Reply:
Thanks for all these review notes!

Regarding this one, in what way do you expect existing python code to break? The texture object will certainly be different but any existing code should remain compatible here since this new field is an addition, no? However admittedly... I modified the python bindings without testing it locally so I'll have to get setup for that.

Reply:
Ok, I updated the relevant python test (in the latest commit) and ran it locally. `getTextures` is in use there.

Reply:
Thanks for the updates @num3ric! Sorry I think my comment was wrongly assuming the Python bindings were returning a Python tuple and we were adding a field in the middle, but we are actually returning a proper aggregate object so my point is not valid!